rhp/v3/encoding.go

```diff
@@ -323,6 +323,9 @@ func (r *RPCExecuteProgramResponse) DecodeFrom(d *types.Decoder) {
 	}
 	(*types.V1Currency)(&r.TotalCost).DecodeFrom(d)
 	(*types.V1Currency)(&r.FailureRefund).DecodeFrom(d)
+	if d.Err() != nil {
+		return // abort before reading output
```

Comment:
The way `SetErr` works it's not necessary to return early here because it clears the buffer. Coincidentally we also check for `err == nil` in the `for` loop in `d.Read`. We save the allocation but I'd still remove it.